lua/customroles/werewolf.lua

```diff
@@ -0,0 +1,797 @@
+            if not ent:IsPlayer() or not ent:IsWerewolf() then return end
+
+            local reduction = werewolf_night_damage_reduction:GetFloat()
+            dmginfo:ScaleDamage(1 - reduction)
+        else
+            local att = dmginfo:GetAttacker()
+            if not IsPlayer(att) or not att:IsWerewolf() then return end
+
+            local penalty = werewolf_day_damage_penalty:GetFloat()
+            dmginfo:ScaleDamage(1 - penalty)
+        end
+    end)
+
+    hook.Add("TTTDrawHitMarker", "Werewolf_TTTDrawHitMarker", function(victim, dmginfo)
+        if not WEREWOLF.isNight then return end
+
+        local reduction = werewolf_night_damage_reduction:GetFloat()
+        if reduction < 1 then return end
+
+        local att = dmginfo:GetAttacker()
+        if not IsPlayer(att) or not IsPlayer(victim) then return end
+
+        if victim:IsWerewolf() then
+            return true, false, true, false
+        end
+    end)
+
+    ----------------------------
+    -- ACTIVE WEREWOLF CHECKS --
+    ----------------------------
+
+    local function CheckForActiveWerewolf()
+        if not timer.Exists("TTTWerewolfTimeChange") then return end
+
+        local werewolf = player.GetLivingRole(ROLE_WEREWOLF)
+        if werewolf then return end
+
+        local night_visibility_mode = werewolf_night_visibility_mode:GetInt()
+        if night_visibility_mode == WEREWOLF_NIGHT_SHOW_IF_HAS_WEREWOLF then
+            timer.Remove("TTTWerewolfTimeChange")
+            if WEREWOLF.isNight then
+                WEREWOLF.ChangeTime(true, false, true)
+            end
+        end
+    end
+
+    hook.Add("PlayerDeath", "Werewolf_PlayerDeath", function(victim, infl, attacker)
+        CheckForActiveWerewolf()
+    end)
+
+    hook.Add("PlayerDisconnected", "Werewolf_PlayerDisconnected", function(ply)
+        CheckForActiveWerewolf()
+    end)
+
+    hook.Add("TTTPlayerRoleChanged", "Werewolf_TTTPlayerRoleChanged", function(ply, oldRole, newRole)
+        CheckForActiveWerewolf()
+
+        if WEREWOLF.isNight then
+            if newRole == ROLE_WEREWOLF and oldRole ~= ROLE_WEREWOLF then
+                local drop_weapons = werewolf_drop_weapons:GetBool()
+                if drop_weapons then
+                    for _, wep in pairs(ply:GetWeapons()) do
+                        local class = WEPS.GetClass(wep)
+                        if class ~= "weapon_zm_improvised" and class ~= "weapon_wwf_claws" and wep.AllowDrop then
+                            ply:DropWeapon(wep)
+                        end
+                    end
+                end
+                ply:Give("weapon_wwf_claws")
+                ply:SelectWeapon("weapon_wwf_claws")
+
+                local transform_model = werewolf_transform_model:GetBool()
+                if transform_model then
+                    oldPlayerModels[ply:SteamID64()] = ply:GetModel()
+                    SetMDL(ply, "models/player/stenli/lycan_werewolf.mdl")
+                end
```

Comment:
These blocks are repeated from above, consider extracting them to a common helper function